test/integration/expression_test.rb

```diff
@@ -29,18 +29,24 @@ def test_float
   def test_range
     assert_equal(1..2, parse_and_eval("(1..2)"))
     assert_equal(3..4, parse_and_eval(" ( 3 .. 4 ) "))
+    assert_equal(0..0, parse_and_eval("('a'..'b')"))
+
+    with_error_mode(:strict) do
+      e = assert_raises(NoMethodError) { parse_and_eval("(1..(1..5))") }
+      assert_match(/undefined method `to_i' for 1..5:Range/, e.message)
```

Comment:
Yeah, this should result in a syntax error. But this PR is bringing behaviour consistency between liquid and liquid-c. I think fixing this belongs to another PR.

Reply:
Fixing liquid-c does belong in another PR, but that needs to happen before testing this case so that we aren't adding a shared test that expects a liquid internal error.

If we add this test first, then fixing liquid-c would cause the test to fail, so adding this test would be a step in the wrong direction.

An alternative would be to make the test more vague, so that it won't prevent us from raising a liquid syntax error.  E.g. use `assert_raises` without the `NoMethodError` argument so it will accept any StandardError, change the block to just parse `"(1..(1..5))"` since we expect the error during parsing and just skip the message assertion for now.  However, that will require 2 follow-up PRs to fix the behaviour in liquid then another to change this test to expect a liquid syntax error.

Reply:
I fixed in 2cf4b1bd55747bdee98ee10b35f7ec8e8eee66d5. Looks like we just need to update `RangeLookup` to use the safer `to_integer` method to convert the range to an integer, which raises `Liquid::ArgumentError` if it can't.

Reply:
I think we actually want to raise a syntax error, so `rescue Liquid::SyntaxError` in the application will handle it along with other syntax errors and because we have exception handling in the library itself to add error context (https://github.com/Shopify/liquid/blob/ed0aebcbc93cade8719da79b72470435c54e938c/lib/liquid/parser_switching.rb#L33-L39).

Besides, we don't even need to handle arbitrary object types like `Utils.to_integer(input)` does.  We can just raise `Liquid::SyntaxError` if the object doesn't `respond_to?(:to_i)`.